### PR TITLE
fix pending when not deploy pilot success then immidate delete istiooperator cr

### PR DESCRIPTION
**Pull request by huanghuangzym**
**Please provide a description of this PR:**
fix for https://github.com/istio/istio/issues/36889

### CODE REVIEW

**Review by Arhell — COMMENTED** [review]
/ok-to-test

**Comment by huanghuangzym** [comment]
> /ok-to-test

can this fix be merged?

File: operator/pkg/helmreconciler/prune.go [hunk 1]
```diff
@@ -116,8 +116,8 @@ func (h *HelmReconciler) PruneControlPlaneByRevisionWithController(iopSpec *v1al
 	}
 	pids, err := proxy.GetIDsFromProxyInfo("", "", iopSpec.Revision, ns)
 	if err != nil {
-		return errStatus,
-			fmt.Errorf("failed to check proxy infos: %v", err)
+		// maybe istiod not run, we should ingnore the err to avoid delete bolcking
+		scope.Errorf("failed to check proxy infos: %v", err)
```
Comment (morvencao) [inline operator/pkg/helmreconciler/prune.go:120]: I don't think it's secure to ignore the error here, at least we should check if the error is no istiod instance running.
We have that type of error, see: https://github.com/istio/istio/blob/0947addd23e3e90dc327ed84494c80c9d1006972/pkg/kube/client.go#L791

Comment (litong01) [reply operator/pkg/helmreconciler/prune.go:120]: I would agree with @morvencao , check the errors if it is indeed, no istiod exists.

Comment (huanghuangzym) [reply operator/pkg/helmreconciler/prune.go:120]: ok  i  will check it ,if this is the error,then ignore it

**Comment by huanghuangzym** [comment]
it seems there is a lot of lint  error  in main branch


pkg/proxy/proxyinfo.go:56:22: ss.ProxyID undefined (type *sidecarSyncStatus has no field or method ProxyID) (typecheck)
				ID:           ss.ProxyID,
				                 ^
pkg/proxy/proxyinfo.go:23:2: "istio.io/istio/pkg/errdict" imported but not used (typecheck)
	"istio.io/istio/pkg/errdict"
	^
pkg/test/framework/components/istio/helm.go:30:2: "istio.io/istio/pkg/test/framework/components/cluster/kube" imported but not used as kubecluster (typecheck)
	kubecluster "istio.io/istio/pkg/test/framework/components/cluster/kube"
	^
pkg/test/framework/components/istio/istio.go:23:2: "istio.io/istio/pkg/test/framework/components/environment/kube" imported but not used (typecheck)
	"istio.io/istio/pkg/test/framework/components/environment/kube"
	^
pkg/test/framework/components/istio/operator.go:44:2: "istio.io/istio/pkg/test/framework/components/cluster/kube" imported but not used as kubecluster (typecheck)
	kubecluster "istio.io/istio/pkg/test/framework/components/cluster/kube"
	^
tests/integration/pilot/common/apps.go:21:2: "strconv" imported but not used (typecheck)
	"strconv"
	^
tests/integration/pilot/common/routing.go:45:2: "istio.io/istio/pkg/test/framework/components/istio/ingress" imported but not used (typecheck)
	"istio.io/istio/pkg/test/framework/components/istio/ingress"
	^
tests/integration/telemetry/stats/prometheus/stats.go:218:2: outputCertAnnot declared but not used (typecheck)
	outputCertAnnot := `
	^
tests/integration/telemetry/stats/prometheus/stats.go:23:2: "strconv" imported but not used (typecheck)
	"strconv"
	^
istioctl/pkg/install/k8sversion/version_test.go:24:2: "k8s.io/client-go/discovery/fake" imported but not used as fakediscovery (typecheck)
	fakediscovery "k8s.io/client-go/discovery/fake"
	^
istioctl/pkg/writer/pilot/status.go:81:30: status.ProxyID undefined (type *writerStatus has no field or method ProxyID) (typecheck)
		if strings.Contains(status.ProxyID, proxyName) {
		                           ^
istioctl/pkg/writer/pilot/status.go:106:20: fullStatus[i].ClusterID undefined (type *writerStatus has no field or method ClusterID) (typecheck)
		if fullStatus[i].ClusterID != fullStatus[j].ClusterID {
		                 ^
istioctl/pkg/writer/pilot/status.go:107:25: fullStatus[i].ClusterID undefined (type *writerStatus has no field or method ClusterID) (typecheck)
			return fullStatus[i].ClusterID < fullStatus[j].ClusterID
			                     ^
istioctl/pkg/writer/pilot/status.go:109:24: fullStatus[i].ProxyID undefined (type *writerStatus has no field or method ProxyID) (typecheck)
		return fullStatus[i].ProxyID < fullStatus[j].ProxyID
		                     ^
istioctl/pkg/writer/pilot/status.go:115:36: status.ClusterSent undefined (type *writerStatus has no field or method ClusterSent) (typecheck)
	clusterSynced := xdsStatus(status.ClusterSent, status.ClusterAcked)
	                                  ^
istioctl/pkg/writer/pilot/status.go:116:37: status.ListenerSent undefined (type *writerStatus has no field or method ListenerSent) (typecheck)
	listenerSynced := xdsStatus(status.ListenerSent, status.ListenerAcked)
	                                   ^
istioctl/pkg/writer/pilot/status.go:117:34: status.RouteSent undefined (type *writerStatus has no field or method RouteSent) (typecheck)
	routeSynced := xdsStatus(status.RouteSent, status.RouteAcked)
	                                ^
istioctl/pkg/writer/pilot/status.go:118:37: status.EndpointSent undefined (type *writerStatus has no field or method EndpointSent) (typecheck)
	endpointSynced := xdsStatus(status.EndpointSent, status.EndpointAcked)
	                                   ^
istioctl/pkg/writer/pilot/status.go:119:20: status.IstioVersion undefined (type *writerStatus has no field or method IstioVersion) (typecheck)
	version := status.IstioVersion
	                  ^
istioctl/pkg/writer/pilot/status.go:124:20: status.ProxyVersion undefined (type *writerStatus has no field or method ProxyVersion) (typecheck)
		version = status.ProxyVersion + "*"
		                 ^
istioctl/pkg/writer/pilot/status.go:127:10: status.ProxyID undefined (type *writerStatus has no field or method ProxyID) (typecheck)
		status.ProxyID, status.ClusterID, clusterSynced, listenerSynced, endpointSynced, routeSynced, status.pilot, version)
		       ^
operator/pkg/util/k8s_test.go:21:2: "k8s.io/client-go/discovery/fake" imported but not used as fakediscovery (typecheck)
	fakediscovery "k8s.io/client-go/discovery/fake"
	^
pilot/pkg/config/kube/crdclient/client_test.go:27:2: "k8s.io/client-go/metadata/fake" imported but not used as metadatafake (typecheck)
	metadatafake "k8s.io/client-go/metadata/fake"
	^
pilot/pkg/credentials/kube/secrets_test.go:25:2: "k8s.io/client-go/kubernetes/fake" imported but not used (typecheck)
	"k8s.io/client-go/kubernetes/fake"
	^
pilot/pkg/xds/fake.go:33:2: "k8s.io/client-go/kubernetes/fake" imported but not used (typecheck)
	"k8s.io/client-go/kubernetes/fake"
	^
pilot/pkg/xds/sds_test.go:28:2: "k8s.io/client-go/kubernetes/fake" imported but not used (typecheck)
	"k8s.io/client-go/kubernetes/fake"
	^
pilot/pkg/xds/eds_sh_test.go:30:2: "istio.io/istio/pilot/pkg/serviceregistry/aggregate" imported but not used (typecheck)
	"istio.io/istio/pilot/pkg/serviceregistry/aggregate"
	^
pkg/kube/client.go:1075:19: undeclared name: `mcs` (typecheck)
	utilruntime.Must(mcs.AddToScheme(scheme))
	                 ^
pkg/test/framework/suite.go:296:24: ctx.Settings undefined (type SuiteContext has no field or method Settings) (typecheck)
		if err != nil && ctx.Settings().CIMode {
		                     ^
pkg/test/framework/suite.go:459:59: ctx.Settings undefined (type SuiteContext has no field or method Settings) (typecheck)
	scopes.Framework.Infof("=== BEGIN: Setup: '%s' ===", ctx.Settings().TestID)
	                                                         ^
pkg/test/framework/suite.go:469:67: ctx.Settings undefined (type SuiteContext has no field or method Settings) (typecheck)
			scopes.Framework.Infof("=== FAILED: Setup: '%s' (%v) ===", ctx.Settings().TestID, err)
			                                                               ^
pkg/test/framework/suite.go:478:63: ctx.Settings undefined (type SuiteContext has no field or method Settings) (typecheck)
	scopes.Framework.Infof("=== DONE: Setup: '%s' (%v) ===", ctx.Settings().TestID, elapsed)
	                                                             ^
pkg/test/framework/components/echo/echotest/filters_test.go:18:2: "strconv" imported but not used (typecheck)
	"strconv"
	^
pkg/test/framework/components/echo/kube/deployment.go:41:2: "istio.io/istio/pkg/test/framework/components/environment/kube" imported but not used (typecheck)
	"istio.io/istio/pkg/test/framework/components/environment/kube"
	^
tests/integration/helm/install_test.go:27:2: "istio.io/istio/pkg/test/framework/components/cluster/kube" imported but not used as kubecluster (typecheck)
	kubecluster "istio.io/istio/pkg/test/framework/components/cluster/kube"
	^
tests/integration/helm/upgrade/util.go:31:2: "istio.io/istio/pkg/test/framework/components/cluster/kube" imported but not used as kubecluster (typecheck)
	kubecluster "istio.io/istio/pkg/test/framework/components/cluster/kube"
	^
tests/integration/pilot/cni_race_test.go:30:2: "istio.io/istio/pkg/kube" imported but not used as istioKube (typecheck)
	istioKube "istio.io/istio/pkg/kube"
	^
tests/integration/pilot/gw_topology_test.go:26:2: "istio.io/istio/pkg/test/framework/components/cluster/kube" imported but not used as kubecluster (typecheck)
	kubecluster "istio.io/istio/pkg/test/framework/components/cluster/kube"
	^
tests/integration/pilot/ingress_test.go:37:2: "istio.io/istio/pkg/test/framework/components/cluster/kube" imported but not used as kubecluster (typecheck)
	kubecluster "istio.io/istio/pkg/test/framework/components/cluster/kube"
	^
tests/integration/pilot/ingress_test.go:39:2: "istio.io/istio/pkg/test/framework/components/environment/kube" imported but not used (typecheck)
	"istio.io/istio/pkg/test/framework/components/environment/kube"
	^
tests/integration/telemetry/outboundtrafficpolicy/helper.go:38:2: "istio.io/istio/pkg/test/framework/components/environment/kube" imported but not used (typecheck)
	"istio.io/istio/pkg/test/framework/components/environment/kube"
	^
tests/integration/telemetry/stats/prometheus/customizemetrics/customize_metrics_test.go:110:2: enableBootstrapDiscovery declared but not used (typecheck)
	enableBootstrapDiscovery := `
	^
tests/integration/telemetry/stats/prometheus/nullvm/dashboard_test.go:31:2: "github.com/prometheus/common/model" imported but not used (typecheck)
	"github.com/prometheus/common/model"
	^
tests/integration/telemetry/tracing/zipkin/clienttracing/client_tracing_test.go:51:7: id declared but not used (typecheck)
						id := uuid.NewString()

**Review by howardjohn — COMMENTED** [review]
maybe should try rebasing - not sure whats going on

**Comment by huanghuangzym** [comment]
> 

ok  i will rebase master  and retry

**Review by morvencao — APPROVED** [review]
thanks!

**Comment by zirain** [comment]
not stale

**Comment by huanghuangzym** [comment]
alreay rebase aganin, can anyone review it ?

**Pull request closed without merging**